providers/class.two-factor-backup-codes.php

```diff
@@ -273,10 +273,12 @@ public function validate_authentication( $user ) {
 	public function validate_code( $user, $code ) {
 		$backup_codes = get_user_meta( $user->ID, self::BACKUP_CODES_META_KEY, true );
 
-		foreach ( $backup_codes as $code_index => $code_hashed ) {
-			if ( wp_check_password( $code, $code_hashed, $user->ID ) ) {
-				$this->delete_code( $user, $code_hashed );
-				return true;
+		if ( is_array( $backup_codes ) && ! empty( $backup_codes ) ) {
```

Comment:
It is possible for the metakey value to come back as 'false' and not an empty array if a specified user has not created any codes in the past.